File: 0683/Sesión 08 - adicional - refactorización/sin_refactorizar/SoftInvModel/src/main/java/pe/edu/pucp/softinv/model/TiposMovimientosDTO.java
package pe.edu.pucp.softinv.model;

public class TiposMovimientosDTO {

    private Integer tipoMovimientoId;
    private String nombre;
    private String descripcion;
    private TipoOperacion tipoOperacion;

    public TiposMovimientosDTO() {
        this.tipoMovimientoId = null;
        this.nombre = null;
        this.descripcion = null;
        this.tipoOperacion = null;
    }
    
    public TiposMovimientosDTO(Integer tipoMovimientoId, String nombre, String descripcion, TipoOperacion tipoOperacion) {
        this.tipoMovimientoId = tipoMovimientoId;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.tipoOperacion = tipoOperacion;
    }

    /**
     * @return the tipoMovimientoId
     */
    public Integer getTipoMovimientoId() {
        return tipoMovimientoId;
    }

    /**
     * @param tipoMovimientoId the tipoMovimientoId to set
     */
    public void setTipoMovimientoId(Integer tipoMovimientoId) {
        this.tipoMovimientoId = tipoMovimientoId;
    }

    /**
     * @return the nombre
     */
    public String getNombre() {
        return nombre;
    }

    /**
     * @param nombre the nombre to set
     */
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    /**
     * @return the descripcion
     */
    public String getDescripcion() {
        return descripcion;
    }

    /**
     * @param descripcion the descripcion to set
     */
    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    /**
     * @return the tipoOperacion
     */
    public TipoOperacion getTipoOperacion() {
        return tipoOperacion;
    }

    /**
     * @param tipoOperacion the tipoOperacion to set
     */
    public void setTipoOperacion(TipoOperacion tipoOperacion) {
        this.tipoOperacion = tipoOperacion;
    }
}

File: 0682/Sesión 12 - WA/con WA/SoftInv/SoftInvModel/TiposMovimientosDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftInv.Model
{
    public class TiposMovimientosDTO
    {
        private int? tipoMovimientoId;
        private string nombre;
        private string descripcion;
        private TipoOperacion? tipoOperacion;

        public TiposMovimientosDTO()
        {
            this.TipoMovimientoId = null;
            this.Nombre = null;
            this.Descripcion = null;
            this.Operacion = null;
        }

        public TiposMovimientosDTO(int tipoMovimientoId, string nombre, string descripcion, TipoOperacion tipoOperacion)
        {
            this.TipoMovimientoId = tipoMovimientoId;
            this.Nombre = nombre;
            this.Descripcion = descripcion;
            this.Operacion = tipoOperacion;
        }

        public int? TipoMovimientoId { get => tipoMovimientoId; set => tipoMovimientoId = value; }
        public string Nombre { get => nombre; set => nombre = value; }
        public string Descripcion { get => descripcion; set => descripcion = value; }
        public TipoOperacion? Operacion { get => tipoOperacion; set => tipoOperacion = value; }
    }
}
